### PR TITLE
Issue #1109: event 発火済み observation AC の評価手順を定義

### LINKED ISSUE #1109: verify: event 発火済み observation AC の評価手順を定義する
## Background

`verify-type: observation` の AC について、**event 発火後にどう評価するか**が `/verify` に定義されていない。仕様と実装が矛盾しており、規定どおりに動くと checkbox は永久に更新されない。

### 矛盾している 2 箇所

**1. `modules/verify-executor.md:244`** — verify-type 解釈テーブル:

> | `observation` | **Skip during normal `/verify` run** — the condition is re-evaluated when the specified `event=<name>` fires via `opportunistic-search.sh --event <name>`. During skip, record as SKIPPED with detail: "observation: waiting for event=<event-name>". |

**2. `scripts/observation-trigger.sh:79`** — event 発火時に対象 Issue へ投稿されるコメント:

```
observation event `auto-run` detected. Run `/verify 797` to verify the condition and update the checkbox.
```

trigger は「`/verify` を実行して checkbox を更新せよ」と案内するが、`/verify` 側の規定は「observation AC は常に SKIPPED」である。**event 発火後に `/verify` を実行しても、executor の規定に従う限り SKIPPED にしかならず checkbox は更新されない。**

「event が既に発火しているので今回は評価する」という分岐がどこにも存在しない。

### 実際に起きた影響

`/auto --batch 1059 1060 1061` 完了時の observation scan が 11 件の Issue にマッチし、その全件に `/verify` をディスパッチした (#797, #839, #841, #843, #984, #995, #1009, #1026, #1027, #1035, #1037)。

このとき各 AC の評価手順が定義されていないため、**11 件すべてで判定ロジックを都度即興で組み立てる**必要があった。具体的には「直近の `/auto` 実行ログを grep する」「`events.jsonl` を集計する」「対象リポジトリの設定・ディレクトリ構成を確認する」といった証拠収集を、AC 文言ごとに個別に設計している。

結果は PASS 1 件 (#1027) / UNCERTAIN 2 件 (#984, #1026) / SKIPPED 8 件。判定自体は可能だったが、手順が標準化されていないため再現性がなく、判定根拠の粒度も実行者依存になる。

### 補足: 即座に判定できるケースがある

#1088 の post-merge observation AC は「`config=` 付き AC が observation scan の対象から除外されることを観察」というもので、**`scripts/opportunistic-search.sh --event auto-run` を read-only で実行するだけで判定できた**。

- 同スクリプトは `observation-trigger.sh` が内部で呼び出す検索そのもの (`observation-trigger.sh:64,66`)
- notification コメントの投稿は `observation-trigger.sh:79` 側にしかなく、検索スクリプトには `gh issue comment` / `gh issue edit` が 1 件も含まれない (grep 一致 0)

このため副作用なしにゲート結果を実測でき、11 件 → 9 件 (#797 が除外) を確認して即 PASS 判定できた。observation 条件が scan 自体の挙動を問う場合に有効な手法だが、これもどこにも記載されていない。

## Purpose

`/verify` に observation AC の評価手順を定義し、event 発火後の `/verify` 実行で checkbox が正しく更新されるようにする。あわせて判定に使える証拠収集手段を列挙し、判定の再現性を確保する。

## 対応方針 (案)

1. **発火済み判定**: Step 4 の comment consumption で拾えるコメント (`observation event \`<name>\` detected`) の有無で、対象 AC の `event=` が既に発火しているかを判定する
2. **評価分岐の追加**: 発火済みの場合は SKIPPED にせず、当該 event 発火時の実行内容と照合して PASS / FAIL / UNCERTAIN を判定する。未発火の場合は従来どおり SKIPPED (`observation: waiting for event=<name>`)
3. **証拠収集手段の列挙**: 判定に使える情報源を明記する
   - 直近の `/auto` 実行ログ (phase 出力・wrapper ログ)
   - `.tmp/auto-events.jsonl` の該当 session_id のイベント
   - read-only な `scripts/opportunistic-search.sh --event <name>` の実行結果 (scan 自体の挙動を問う条件の場合)
   - 対象リポジトリの設定・ファイル構成 (`.wholework.yml`、ディレクトリ有無など、条件の前提が成立するか)
4. **前提不成立時の扱い**: 観測対象の状況自体が発生していない場合 (例: `always-pr` 未設定のリポジトリで `always-pr: true` 下の挙動を観察する条件) は SKIPPED とし、その理由を `## Acceptance Test Results` の Details 列に記録する (既存の SKIPPED 詳細記述パターン `modules/verify-executor.md:244` の "detail:" 表記に合わせる)
5. `modules/verify-executor.md:244` の記述を上記分岐と整合する形に更新する

### Pre-merge (auto-verified)

- [x]  event 発火済み observation AC の評価分岐が追加されている
- [x]  証拠収集手段が列挙されている
- [x]  `verify-executor.md` の記述が新しい分岐と整合している
- [x]  `tests/verify.bats` が PASS する

### Post-merge

- [x] observation AC を持つ Issue に対して event 発火後に `/verify` を実行し、条件が満たされている場合に checkbox が更新され `phase/done` へ遷移することを確認する 

## Auto-Resolved Ambiguity Points

- **「対応方針 (案)」4. の記録先を「Notes」から「`## Acceptance Test Results` の Details 列」へ明確化** — 既存の `/verify` 出力フォーマット (`skills/verify/SKILL.md` Step 9 の Auto Verification テーブル、`modules/verify-executor.md:244` の "detail:" 表記) には "Notes" という項目が存在せず、既存の SKIPPED 理由記録パターンと整合する呼称に統一した。AC の判定内容には影響しない。

## Related

- #1088 — `config=` ゲートの実装 Issue。post-merge observation AC を read-only な `opportunistic-search.sh` で即座に PASS 判定できた実例
- #1026 — observation AC への notification 蓄積問題 (本 Issue は蓄積ではなく「発火後に評価できない」側)
- #1099 — `observation-trigger.sh` の重複投稿抑止
- `modules/verify-executor.md:244` / `scripts/observation-trigger.sh:79` — 矛盾している 2 箇所

### PR DESCRIPTION
**Pull request by saito**
Closes #1109

## 概要

`verify-type: observation` の AC について、**event 発火後にどう評価するか**が `/verify` に定義されておらず、仕様と実装が矛盾していた問題を解消する。

- `modules/verify-executor.md:244` — observation を「**Skip during normal `/verify` run**」と規定し、常に SKIPPED として記録するよう指示
- `scripts/observation-trigger.sh:79` — event 発火時に「Run `/verify N` to verify the condition and update the checkbox」というコメントを投稿

規定どおりに動くと、event 発火後に `/verify` を実行しても checkbox は永久に更新されない。

Spec: `docs/spec/issue-1109-verify-observation-eval.md`

## 変更内容

| ファイル | 変更 |
|---|---|
| `modules/verify-executor.md` | verify-type 表の `observation` 行を「常に SKIPPED」から「event 発火済みなら評価 / 未発火なら従来どおり SKIPPED」の分岐に変更。評価手順は `skills/verify/SKILL.md` Step 8c を参照 |
| `skills/verify/SKILL.md` Step 4 | post-merge 条件の振り分けに observation 専用の分岐 (Step 8c へのルーティング) を追加 |
| `skills/verify/SKILL.md` Step 7 | Post-merge Briefing の "Claude Executable?" 列を発火済み/未発火で出し分ける記述に更新 |
| `skills/verify/SKILL.md` Step 8c (新設) | 発火判定 → 証拠収集 → PASS/FAIL/UNCERTAIN/SKIPPED 判定 |
| `skills/verify/SKILL.md` Post-Step 8 | checkbox flip の PASS 判定元に Step 8c を追加 |
| `skills/verify/SKILL.md` Step 11(a) | reopen 判定対象に Step 8c で PASS/FAIL 確定した observation 条件を追加 |

### Step 8c の設計

**発火判定**: Step 4 の comment consumption cutoff の影響を受けないよう、Issue の全コメント履歴を直接検索する。

**証拠収集 (best-effort)**: 直近の `/auto` 実行ログ / `.tmp/auto-events.jsonl` / read-only な `opportunistic-search.sh --event <name>` の再実行 / 対象リポジトリの `.wholework.yml` とファイル構成。

`opportunistic-search.sh` は `observation-trigger.sh` が内部で呼ぶ検索そのもので、コメント投稿は trigger 側にしかないため副作用なく再実行できる。

**判定**: 条件充足→PASS / 証拠が矛盾→FAIL / 不十分→UNCERTAIN / **前提自体が当該リポジトリで成立しない**(参照する `config=` キーが未設定、参照するディレクトリ・機能が存在しない等)→SKIPPED。

## 検証

- `bats tests/verify.bats tests/verify-executor.bats tests/observation-trigger.bats tests/opportunistic-search.bats tests/verify-heuristics.bats tests/verify-rubric.bats tests/run-verify.bats tests/verify-dirty-detection.bats` — **90 件すべて PASS** (`not ok` 0 件)
- `python3 scripts/validate-skill-syntax.py skills/verify/SKILL.md` — 0 error / 0 warning
- `bash scripts/check-forbidden-expressions.sh` — 違反なし

## 注記: code フェーズの silent no-op からの復旧

このブランチの実装コミットは、`/auto 1109` の code フェーズが **3 回連続で silent no-op に陥った**あと、親セッションが復旧として作成したものである。

`/code` セッションが headless (`claude -p`) で bats テストスイートをバックグラウンド実行し、届かない完了通知を待ってターンを終える — **#1097 として起票済みのパターン**そのものが再現した。

```
長時間実行タスクとしてバックグラウンドに移行した bats テストスイートの完了を待ちます。
Warning: claude exited 0 but code-pr phase did not complete (silent no-op).
auto-retry: code phase silent no-op, retry 1/3
...
auto-retry: code phase silent no-op, retry 2/3
```

3 回目は並行セッションが main に未コミット変更を持っていたため `check-verify-dirty` のガードで停止した (このガード自体は並行セッションの作業を保護する正しい挙動)。

親セッションによる復旧の手順:

- worktree の変更ファイル 3 件が Spec の `## Changed Files` と完全一致することを確認
- Spec の Implementation Steps 5 箇所すべてが実装されていることを diff で確認
- worktree 内で上記の検証 (bats 90 件・syntax validator・forbidden expressions) を実行し全 green を確認
- そのうえで commit → `origin/main` へ rebase → push → PR 作成

`/code` 自身の commit ステップは実行されていないため、`## Code Retrospective` は存在しない。`/review` では diff と Spec の照合を通常より丁寧に行うことを推奨する。

### CODE REVIEW

File: skills/verify/SKILL.md [hunk 1]
```diff
@@ -375,9 +376,40 @@ If "Claude Execute" is selected: run the command/approach → judge PASS or FAIL
...
+**1. Fired detection**
+
+Extract `<name>` from the `event=<name>` attribute. Fired status is not subject to Step 4's comment-consumption cutoff — search the Issue's full comment history directly:
+
+```bash
+gh issue view "$NUMBER" --json comments --jq '.comments[].body' \
+  | grep -F "observation event" | grep -F "detected" | grep -F -- "$EVENT_NAME"
+```
+
+- **No match**: the event has not fired yet. Record as SKIPPED with detail: "observation: waiting for event=<event-name>" (unchanged from prior behavior). Do not proceed to evidence collection.
```
Comment (saito) [inline skills/verify/SKILL.md]: [Edge Case/Robustness] SHOULD: If `gh issue view` itself fails (auth/network/rate-limit), the pipe yields no output and this falls through to "No match: not fired yet" — silently recording SKIPPED even though the real cause is an inability to check, not an unfired event. Other `gh`-based checks in this file document explicit fail-open/fail-closed semantics; Step 8c has no equivalent guard. Recommend an explicit branch: if the `gh issue view` command fails (non-zero exit / no stdout), record UNCERTAIN ("could not confirm fired status (gh error)") instead of SKIPPED.

File: skills/verify/SKILL.md [hunk 2]
```diff
@@ -375,9 +376,40 @@ If "Claude Execute" is selected: run the command/approach → judge PASS or FAIL
...
+
+For each unchecked post-merge condition marked `<!-- verify-type: observation event=<name> -->`:
+
+**1. Fired detection**
+
+Extract `<name>` from the `event=<name>` attribute. Fired status is not subject to Step 4's comment-consumption cutoff — search the Issue's full comment history directly:
+
+```bash
+gh issue view "$NUMBER" --json comments --jq '.comments[].body' \
+  | grep -F "observation event" | grep -F "detected" | grep -F -- "$EVENT_NAME"
```
Comment (saito) [inline skills/verify/SKILL.md]: [Edge Case/Robustness] SHOULD: `grep -F -- "$EVENT_NAME"` is an unanchored substring match against the comment body. If a future observation event name is a substring of another (e.g. `event=auto` vs. `event=auto-run`), a fired notification for the longer name would also match the grep for the shorter, still-unfired name, causing Step 8c to evaluate the wrong AC as fired. Not an active bug with today's event names (`pr-review-full`/`pr-review-light`/`auto-run`/`watchdog-kill`/`fix-cycle` — none is a substring of another), but latent given this mechanism is designed for arbitrary future event names. Recommend anchoring the match to the exact backtick-quoted token, e.g. `` grep -F -- "\`${EVENT_NAME}\`" `` (matching `scripts/observation-trigger.sh:79`'s literal formatting).

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| Step 8c に発火済み observation AC の評価分岐が追加されている | PASS | `skills/verify/SKILL.md` に "Step 8c: Observation Post-merge Conditions" が新設され、fired detection → evidence collection → PASS/FAIL/UNCERTAIN/SKIPPED 判定の流れが明記されている |
| 証拠収集手段が複数列挙されている (auto 実行ログ / auto-events.jsonl / opportunistic-search.sh) | PASS | Step 8c の "2. Evidence collection" に4種の証拠源 (auto実行ログ、`.tmp/auto-events.jsonl`、`opportunistic-search.sh --event` read-only 再実行、`.wholework.yml`/ディレクトリ構成) が列挙されている |
| `verify-executor.md` の observation 行が発火状況に応じた分岐に更新されている | PASS | `modules/verify-executor.md:244` の observation 行が「常に SKIPPED」から「発火済みなら Step 8c へ / 未発火なら SKIPPED」の分岐記述に更新されている |
| `tests/verify.bats` が PASS する | PASS | CI ジョブ `Run bats tests` が SUCCESS (CI 参照フォールバック) |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### General Comments (issues where path/line cannot be identified)
- **docs/spec/issue-1109-verify-observation-eval.md:28**: [Spec Deviation] SHOULD: The Spec's steering-doc sync audit doesn't cover `modules/observation-trigger.md` or `modules/verify-classifier.md`, both of which still assert the exact pre-fix behavior this PR corrects ("not verified during a normal `/verify` run" with no fired/unfired branch). Recommend extending the audit to include these two files and updating their wording to match the branch already applied to `modules/verify-executor.md`.
- **modules/observation-trigger.md:7**: [Documentation Consistency] SHOULD: This line (and `modules/verify-classifier.md:30-31`, same wording) still states "not verified during a normal `/verify` run; instead, it is re-evaluated automatically when the specified event fires" — this is exactly the pre-fix behavior Issue #1109 identifies as the bug. After this PR, once the event has fired, the AC IS evaluated during a normal `/verify` run (Step 8c). Recommend updating both files to note the fired/unfired branch, consistent with the wording already applied to `modules/verify-executor.md`.

**[Edge Case/Robustness] `tests/verify.bats` — 新設された Step 8c 分岐の回帰テストが未追加**
Pre-merge AC の `bats tests/verify.bats` は PASS しているが、`tests/verify.bats` / `tests/verify-executor.bats` のいずれにも `observation` 関連のテストケースが存在しない (grep 0件確認済み)。テストスイートの PASS は新しい Step 8c ロジックを実際には検証していない。Issue の Type=Bug であることを踏まえ、`skills/verify/SKILL.md` に "Step 8c: Observation Post-merge Conditions" セクションが存在すること、および `modules/verify-executor.md` の observation 行が発火/未発火分岐を反映していることを確認する bats ケースの追加を推奨する (`tests/verify.bats` の既存の Step 5 系プローズアサーションと同じパターン)。
Severity: SHOULD

### Issue Count Summary

- Posted as line comments: 4 (MUST: 0, SHOULD: 4, CONSIDER: 0)
- General Comments (in body): 1 (SHOULD: 1)

---
Next: run `/merge 1121`.

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Documentation Consistency | modules/observation-trigger.md / modules/verify-classifier.md | 7 / 30-31 | Both files still described the pre-fix "always not verified" behavior, contradicting the new fired/unfired branch | Resolved | Updated both files' wording to note the fired/unfired branch, consistent with `modules/verify-executor.md` |
| Edge Case/Robustness | skills/verify/SKILL.md | 392-393 | `gh issue view` failure fell through to the same branch as "not fired yet," silently recording SKIPPED | Resolved | Added an explicit UNCERTAIN branch when the `gh issue view` command itself fails |
| Edge Case/Robustness | skills/verify/SKILL.md | 389 | Unanchored `grep -F -- "$EVENT_NAME"` substring match risked matching a longer event name's fired notification as a substring of a shorter, still-unfired event name | Resolved | Anchored the match to the exact backtick-quoted event token as posted by `scripts/observation-trigger.sh:79` |
| Edge Case/Robustness (regression test) | tests/verify.bats | — | No test coverage for the new Step 8c branch or the updated `verify-executor.md` observation row | Resolved | Added 7 new bats cases covering fired/unfired branching, PASS/FAIL/UNCERTAIN/SKIPPED judgment, evidence collection sources, gh-failure handling, anchored matching, and the `verify-executor.md` row update |
| Spec Deviation | docs/spec/issue-1109-verify-observation-eval.md | 28 | Spec's steering-doc sync audit didn't cover `modules/observation-trigger.md` / `modules/verify-classifier.md` | Skipped | Spec files are disposable per project convention; the underlying documentation gap is resolved by the Documentation Consistency fix above, so no further Spec edit was made |

### Lightweight Re-check Results

- `bats tests/verify.bats tests/verify-executor.bats tests/observation-trigger.bats tests/opportunistic-search.bats tests/verify-heuristics.bats tests/verify-rubric.bats tests/run-verify.bats tests/verify-dirty-detection.bats` — 97/97 PASS (90 original + 7 new)
- `python3 scripts/validate-skill-syntax.py skills/verify/SKILL.md` — 0 error / 0 warning (an initial fix draft using a double-backtick inline-code span tripped the validator's naive stripping regex over a large span of the document; rewritten to avoid nested backticks, now clean)
- `bash scripts/check-forbidden-expressions.sh` — no violations
- New issues: none
